### PR TITLE
Appeal list contact row functionality

### PR DESCRIPTION
**Pull request by dr-bizz**
## Description
In this PR, I added functionality to the contact row. Allow the user to add a pledge, edit a pledge, delete a pledge, remove a contact from an appeal and add an excluded contact.

You will see I added some files from the commit `[1. List Filters - Add contacts to appeal](https://github.com/CruGlobal/mpdx-react/commit/c2412ee8379da48151c5e0cdfc13e81d0862fd3c)` otherwise this PR would break.  This should be merged in before this merges so I will remove that once that happens.

### Changes
- **Add a Pledge:** Implemented the ability to add a new pledge with associated donation amounts.
- **Edit a Pledge:** Enabled the editing of existing pledges within a contact row.
- **Delete a Pledge:** Added functionality to delete a pledge.
- **Remove Contact from Appeal:** Users can now remove a contact from an appeal.
- **Add Excluded Contact:** Implemented the option to add contacts to the excluded list.

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

**Comment by dr-bizz** [comment]
If `codecov/patch`  is above 80% I have been approving PRs as that is a pass in my book.

**Review by canac — CHANGES_REQUESTED** [review]
I haven't tested it thoroughly yet but here's my comments so far.

**Comment by dr-bizz** [comment]
@canac I can't see you comments

**Review by canac — CHANGES_REQUESTED** [review]
Looking great! Just some minor suggestions.

File: src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx [hunk 1]
```diff
@@ -1,54 +1,130 @@
...
+const ContactRowActions = styled(Box)(() => ({
+  opacity: 0,
+  transition: 'opacity 0.3s',
+}));
+
+type FormatPledgeOrDonationProps = {
+  amount?: number | null;
+  currency?: string | null;
+  appealStatus: AppealStatusEnum;
+  date?: any;
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx:68]: Renamed to `dateOrFrequency`

File: src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx [hunk 2]
```diff
@@ -60,73 +136,318 @@ export const ContactRow: React.FC<Props> = ({ contact, useTopMargin }) => {
...
+      // Currently we grab all the donations and filter them by the appeal id
+      // We need a query that allows us to filter by the appeal id
+      // Maybe buy the backend team some donuts and ask them to add a filter to the donations query
+      const appealDonations = donations.nodes.filter(
+        (donation) => donation?.appeal?.id === appealId,
+      );
+
+      const givenDonations = appealDonations.map((donation) => {
+        const donationAmount =
+          donation?.appealAmount?.amount &&
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx:208]: I've made it simplier

File: src/components/Tool/Appeal/Modals/DeleteAppealContact/DeleteAppealContactModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+const handleClose = jest.fn();
+const mutationSpy = jest.fn();
+const refetch = jest.fn();
+
+interface ComponentsProps {
+  contact?: AppealContactInfoFragment;
+}
+
+const Components = ({}: ComponentsProps) => {
+  let cardinality = 0;
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/DeleteAppealContact/DeleteAppealContactModal.test.tsx:47]: For sure! Whatever makes it easier to understand

File: src/components/Tool/Appeal/Modals/DeletePledgeModal/DeletePledgeModal.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [deleteAccountListPledge] = useDeleteAccountListPledgeMutation();
+  const [mutating, setMutating] = React.useState(false);
+  const { contactsQueryResult } = React.useContext(
+    AppealsContext,
+  ) as AppealsType;
+
+  const handleConfirm = async () => {
+    setMutating(true);
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/DeletePledgeModal/DeletePledgeModal.tsx:46]: I love this feature, can't believe I know about it before

File: src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.tsx [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+          <form onSubmit={handleSubmit}>
+            <DialogContent>
+              <Grid item>
+                <ContactsAutocomplete
+                  accountListId={accountListId ?? ''}
+                  value={[contactId]}
+                  onChange={(contactId) => {
+                    setFieldValue('contactId', contactId);
+                  }}
+                  disabled={true}
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.tsx:214]: Great point!

File: src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.tsx [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+                              PaperProps: {
+                                style: {
+                                  maxHeight: '300px',
+                                  overflow: 'auto',
+                                },
+                              },
+                            }}
+                            error={!!errors.status && touched.status}
+                          >
+                            <MenuItem value={''} disabled></MenuItem>
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/PledgeModal/PledgeModal.tsx:387]: Removed blank entry

File: src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx [hunk 7]
```diff
@@ -204,20 +214,14 @@ export const ContactRow: React.FC<Props> = ({
...
-            : donation?.appealAmount?.amount ||
-              currencyFormat(
-                0,
-                donation?.appealAmount?.convertedCurrency,
-                locale,
-              );
+        const amount = donation?.appealAmount?.amount;
+        const currency = donation?.appealAmount?.convertedCurrency;
+        let donationAmount = '';
+        if (amount && currency) {
```
Comment (canac) [inline src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx:220]: I didn't realize what you were trying to accomplish was so simple. I thought there was a branch where you needed to render the raw amount without calling currency format. Thanks for making the logic verbose and clear! If you wanted to simplify, you could do `const donationAmount = currencyFormat(amount && currency ? amount : 0, currency, locale);`

**Review by canac — APPROVED** [review]

**Pull request merged**